src/sortStudents.ts

```diff
@@ -1,16 +1,84 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number []
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverage(studentAverages: number[]): number {
+  return studentAverages.reduce((prevResult, currentGrade) => {
+    return (prevResult + currentGrade);
+  }, 0) / studentAverages.length;
+}
+
+export function sortStudents(
+  students: Student [],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const studentsCopy = [...students];
+  const sortOrder: 1 | -1 = order === 'asc' ? 1 : -1;
+
+  switch (sortBy) {
+    case SortType.Name:
+      if (order === 'desc') {
+        studentsCopy.sort((stud1, stud2) => stud2.name
+          .localeCompare(stud1.name));
+      }
+
+      studentsCopy.sort((stud1, stud2) => stud1.name
+        .localeCompare(stud2.name));
+
+      break;
+
+    case SortType.Surname:
+      if (order === 'desc') {
+        studentsCopy.sort((stud1, stud2) => stud2.surname
+          .localeCompare(stud1.surname));
+      }
+
+      studentsCopy.sort((stud1, stud2) => stud1.surname
+        .localeCompare(stud2.surname));
+
+      break;
+
+    case SortType.Married:
+      studentsCopy.sort((stud1, stud2) => (+stud1.married
```

Comment:
Number() is more obvious way
```suggestion
      studentsCopy.sort((stud1, stud2) => (Number(stud1.married)
```